src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,10 +1,52 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isRequired = () => {
+    if (title.trim() && imgUrl.trim() && imdbUrl.trim() && imdbId.trim()) {
+      return false;
+    }
+
+    return true;
+  };
```

Comment:
```suggestion
  const isFormValid = Boolean(title.trim() && imgUrl.trim() && imdbUrl.trim() && imdbId.trim());
```
Just store value in a variable, you don't need to create a function for this.
And the name `isRequired` doesn't match what this function does